mystery_o_matic/output/latex/utils.py

```
@@ -0,0 +1,118 @@
+
+def save_tex(outdir, language, tex):
+    output_dir = f"{outdir}/{language}"
+    with suppress(OSError):
+        makedirs(output_dir)
+
+    filename = f"{output_dir}/mystery.tex"
+    try:
+        with open(filename, "w", encoding='utf-8') as f:
+            f.write(tex)
+    except OSError as e:
+        raise RuntimeError(f"Failed to save LaTeX file: {e}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)